app/src/main/java/com/infomaniak/euria/upload/UploadManager.kt

```diff
@@ -71,52 +73,104 @@ class UploadManager @Inject constructor(
+        withValidOrganizationId(webView) { organizationId ->
+            val currentUser = requestCurrentUser() ?: return@withValidOrganizationId
+            // First loop to get all files information to send it to the WebView
+            var filesInfo = getFilesInfo(uris)
+
+            // Send the `prepareFilesForUpload` with all files in order to display the right number of elements in the WebView
+            // but also to let the WebView display errors if one file is not valid. This JS method will also return
+            // an array of UUIDs representing the files that are valid.
+            filesInfo = prepareFilesForUpload(webView, filesInfo)
+
+            // If no files are valid, we can leave safely
+            if (filesInfo.isEmpty()) return@withValidOrganizationId
+
+            val okHttpClient = getHttpClient(currentUser.apiToken.accessToken)
+
+            filesInfo.forEach { fileInfo ->
+                if (fileInfo.uri == null) return@forEach
+
+                // Creating deferred to upload files to make it easier to cancel an upload
+                uploadJobs[fileInfo.localId] = async(Dispatchers.IO, start = CoroutineStart.LAZY) {
+                    startFileUpload(
+                        uri = fileInfo.uri,
+                        okHttpClient = okHttpClient,
+                        fileInfo = fileInfo,
+                        organizationId = organizationId,
+                        jsonParser = jsonParser,
+                        webView = webView
+                    )
+                }
+            }
+            startUploadFiles()
         }
-        startUploadFiles()
     }
 
     fun cancelUpload(localId: String) {
         uploadJobs[localId]?.cancel()
     }
 
+    private suspend fun startBitmapUpload(
+        bitmap: Bitmap,
+        okHttpClient: OkHttpClient,
+        fileInfo: FileInfo,
+        organizationId: String,
+        jsonParser: Json,
+        webView: WebView?,
+    ) {
+        try {
+            val byteArrayOutputStream = ByteArrayOutputStream()
+            bitmap.compress(Bitmap.CompressFormat.JPEG, 100, byteArrayOutputStream)
+
+            val uploadFileResponse = uploadFile(
+                info = fileInfo,
+                byteArray = byteArrayOutputStream.toByteArray(),
+                organizationId = organizationId,
+                okHttpClient = okHttpClient,
+            )
+
+            currentCoroutineContext().ensureActive()
+
+            sendUploadResultToWebView(
+                result = uploadFileResponse,
+                jsonParser = jsonParser,
+                fileInfo = fileInfo,
+                webView = webView,
+            )
+        } catch (e: CancellationException) {
+            throw e
+        } catch (_: SocketTimeoutException) {
+            sendFileUploadError(fileInfo, "", webView, jsonParser)
+        } catch (_: Exception) {
+            sendFileUploadError(fileInfo, "", webView, jsonParser)
+        }
```

Comment:
This is useless I think

You can use a runCatching.cancellable, then direclty call this sendFileUploadError